app/assets/javascripts/govuk_publishing_components/components/accordion.js

```diff
@@ -55,6 +56,12 @@ window.GOVUK.Modules.GovukAccordion = window.GOVUKFrontend.Accordion;
         console.error('Could not read accordion data attributes error: ' + e.message, window.location)
       }
     }
+
+    var showAllAttributesGa4 = this.$module.getAttribute('data-ga4-show-all-attributes')
+    if (showAllAttributesGa4) {
+      showAll = this.$module.querySelector(this.showAllControls)
+      showAll.setAttribute('data-ga4-event', showAllAttributesGa4)
```

Comment:
Instead of passing these attributes to the component and then moving them onto the showall button, couldn't we just hardcode them? Something like this:

```Javascript
// look for data attributes to put onto the 'show/hide all' link
var showAll = this.$module.querySelector(this.showAllControls)
var showAllAttributes = this.$module.getAttribute('data-show-all-attributes')
if (showAllAttributes) {
  try {
    var values = JSON.parse(showAllAttributes)
    var keys = Object.keys(values)
    for (var i = 0; i < keys.length; i++) {
      showAll.setAttribute('data-' + keys[i], values[keys[i]])
    }
  } catch (e) {
    console.error('Could not read accordion data attributes error: ' + e.message, window.location)
  }
}

// if GA4 is enabled, set attributes on 'show all sections' for tracking using ga4-event-tracker
var isGa4Enabled = this.$module.hasAttribute('data-ga4-expandable')
if (isGa4Enabled) {
  var indexTotal = this.$module.querySelectorAll('.govuk-accordion__section').length
  var showAllAttributesGa4 = {"event_name":"select_content","type":"accordion","index":0,"index_total": indexTotal}
  showAll.setAttribute('data-ga4-event', JSON.stringify(showAllAttributesGa4))
}
```

Reply:
Thanks @andysellick - I've added this change and then will move it to `ga4-core.js` once the link tracker PR is merged.

Reply:
Does this still need moving to `ga4-core.js`? I can't think what else will need this right now. Maybe move it later when we need it, shouldn't have to block this PR.

Reply:
This is due to `ga4-core` not being initialised in `govuk_publishing_components`, so the accordion would crash on the developer frontend which showcases the component. Therefore we decided to share the function at a later date.